.github/workflows/ci-unix-shared-installed.yml

```diff
@@ -14,15 +14,19 @@ concurrency:
 
 jobs:
   build-shared-installed:
-    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-24.04, macos-latest]
+        oldest-cmake: [false, true]
         include:
-          - runs-on: ubuntu-24.04
+          - os: ubuntu-24.04
             compiler: gcc
             gcc: 14
+        exclude:
+          - os: macos-latest
+            oldest-cmake: true
```

Comment:
Optional: I can't quite figure out what the `include` and `exclude` do. A comment for each of them may be useful.

Reply:
Rules are indeed tricky: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow . Done